Azure/azure-cli
# [Core] Add new get command loader mechanism when loading CommandLoader

**Description**  

 
According to spec[ design](https://microsoft.sharepoint.com/:w:/t/AzureManagementExperience/EZ9Zu0EaSq9OvLQgJDVcaj0BRs9ewFhKMAy5HIFjjKmBEw?e=M0suFg), two kinds of CommandLoader will be introduced in storage module.

Here in storage initialization, AzureStackStorageCommandLoader will be loaded when cloud profile is for Azure Stack and StorageCommandLoader will be used in public cloud.
For other modules, if there is requirement to load different CommandLoader, the `get_command_loader()` function can be utilized.

Here `command_v2.py` and `params_v2.py` are totally the same as existing `command.py` and `params.py`.

**Testing Guide**  

In public cloud, `az storage -h` will see all commands for public cloud.

Switch the profile with `az cloud set`, `az storage -h` will only show commands Azure Stack supports.

**History Notes**  


[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.
